tests/integration/validation.js

```diff
@@ -0,0 +1,100 @@
+'use strict'
+
+const request = require('supertest')
+const tap = require('tap')
+
+const port = 13003
+process.env.SERVER_PORT = port
+
+const {withTestMapperServer, waitForURLReachable} = require('../utils')
+
+tap.test(
+  'ValidationMiddleware',
+  withTestMapperServer(port, async t => {
+    t.test(
+      'validateBodyMiddleware should validate lookupRequests and requestBody data',
+      async t => {
+        const testEndpoint = {
+          name: 'Test Endpoint',
+          endpoint: {
+            pattern: '/validationTest'
+          },
+          transformation: {
+            input: 'JSON',
+            output: 'JSON'
+          },
+          requests: {
+            lookup: [
+              {
+                id: 'checkTestServerUp',
+                config: {
+                  method: 'get',
+                  url: `http://localhost:${port}/_health`
+                }
+              }
+            ]
+          },
+          inputValidation: {
+            type: 'object',
+            properties: {
+              lookupRequests: {
+                type: 'object',
+                properties: {
+                  checkTestServerUp: {
+                    type: 'object',
+                    properties: {
+                      status: {
+                        type: 'string',
+                        enum: ['UP']
+                      }
+                    },
+                    required: ['status']
+                  }
+                },
+                required: ['checkTestServerUp']
+              },
+              requestBody: {
+                type: 'object',
+                properties: {
+                  name: {
+                    type: 'string'
+                  }
+                },
+                required: ['name']
+              }
+            },
+            required: ['lookupRequests', 'requestBody']
+          }
+        }
+
+        await request('http://localhost:13003')
+          .post('/endpoints')
+          .send(testEndpoint)
+          .set('Content-Type', 'application/json')
+          .expect(201)
+
+        // The new endpoint may take a few milliseconds to trigger a cache update.
+        // If the read endpoint comes back positive then the cache should be updated and we can safely post.
+        await waitForURLReachable(
+          `http://localhost:13003/endpoints?name=Test Endpoint`,
+          500,
+          5
+        ).catch(error => {
+          t.fail(`Should not error. Caused by: ${error.message}`)
+        })
```

Comment:
@MattyJ007 I think we should take this code out of this module. We can have in the utils module and wrap the logic into one function that is generic and can be used to create the test endpoints. The function will take the endpoints object as an arg

Reply:
I think that is an over abstracting the test.
Part of a good test is having all the test variables available within the test - as far as is feasible.
I think it's useful to have more control and insight into the endpoint creation. It's not a lot of code and moving it into the test utils file makes our tests more difficult to read.